File: backend/app/schemas/agents.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class AgentRunOut(BaseModel):
    id: str
    agent_name: str
    agent_kind: str
    provider: str
    model: str
    entity_type: str | None = None
    entity_id: str | None = None
    status: str
    outcome: str | None = None
    fallback_reason: str | None = None
    confidence: str | None = None
    cost_usd: str
    latency_ms: int
    auto_completed: bool
    created_at: datetime


class AgentRunsResponse(BaseModel):
    runs: list[AgentRunOut]


class AgentRollupRowOut(BaseModel):
    agent_name: str
    run_count: int
    total_cost_usd: str
    fallback_rate: str
    auto_count: int
    escalated_count: int


class AgentRollupResponse(BaseModel):
    window: str
    generated_at: datetime
    agents: list[AgentRollupRowOut]

File: frontend/src/lib/agents.ts
import { authHeaders } from "@/lib/authFetch";

const API_URL = process.env.NEXT_PUBLIC_API_URL || "http://127.0.0.1:8000";

export interface AgentRun {
  id: string;
  agent_name: string;
  agent_kind: string;
  provider: string;
  model: string;
  entity_type: string | null;
  entity_id: string | null;
  status: string;
  outcome: string | null;
  fallback_reason: string | null;
  confidence: string | null;
  cost_usd: string;
  latency_ms: number;
  auto_completed: boolean;
  created_at: string;
}

export interface AgentRunsResponse {
  runs: AgentRun[];
}

export async function fetchAgentRuns(): Promise<AgentRunsResponse> {
  const res = await fetch(`${API_URL}/api/agents/runs`, { headers: authHeaders() });
  if (!res.ok) throw new Error(`agents/runs ${res.status}`);
  return res.json();
}
